### PR TITLE
Optimisation babel transform + fixes

### PR DESCRIPTION
**Pull request by zamotany**
* Babel transform to strip unused `Platform.select` platform code when creating static bundle
* Remove unnecessary `__webpack_require__.p` from async chunk loading code
* Fix response from debugger worker when communicating through bridge

### CODE REVIEW

File: packages/haul-babel-preset-react-native/src/transforms/__tests__/stripDeadPlatformSelect.test.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  it('should replace Platform.select with value for default case if there is no matching platform', () => {
+    const { code } = transformSync(fixture1, {
+      plugins: [[stripDeadPlatformSelect, { platform: 'custom' }]],
+      babelrc: false,
+      configFile: false,
+    }) || { code: '' };
+
+    expect(code).toMatch(/const value = {\s*test: true\s*}/);
+    expect(code).not.toMatch('Platform.select');
+    expect(code).toMatchSnapshot();
```
Comment (thymikee) [inline packages/haul-babel-preset-react-native/src/transforms/__tests__/stripDeadPlatformSelect.test.ts]: I'd use `toMatchInlineSnapshot` but whatever

File: packages/haul-babel-preset-react-native/src/transforms/__tests__/stripDeadPlatformSelect.test.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+const fixture6 = `
+const value = Platform.select({
+  ios: false,
+  android() {}
+});
+`;
+
+describe('stripDeadPlatformSelect transform', () => {
+  it('should replace Platform.select with value for default case if there is no matching platform', () => {
+    const { code } = transformSync(fixture1, {
```
Comment (thymikee) [inline packages/haul-babel-preset-react-native/src/transforms/__tests__/stripDeadPlatformSelect.test.ts]: also grouping fixtures is harder for readability, can you move them inside the test cases?

File: packages/haul-babel-preset-react-native/src/index.ts [hunk 3]
```diff
@@ -30,7 +30,16 @@ export default function getHaulBabelPreset() {
         plugins: [require('@babel/plugin-transform-flow-strip-types')],
       },
       {
-        plugins: defaultPlugins,
+        plugins: defaultPlugins.concat(
+          process.env.HAUL_PLATFORM
```
Comment (thymikee) [inline packages/haul-babel-preset-react-native/src/index.ts]: Can you make `platform` a preset config? This is hacky

File: packages/haul-core/assets/debuggerWorker.js [hunk 4]
```diff
@@ -117,16 +117,18 @@ class DebuggerWorker {
...
     } catch (e) {
+      error = e;
       if (self.ErrorUtils) {
         self.ErrorUtils.reportFatalError(e);
       } else {
         console.error(e);
       }
     } finally {
-      sendReply(null, null);
+      sendReply(null, error);
```
Comment (thymikee) [inline packages/haul-core/assets/debuggerWorker.js]: I guess it doesn't matter if error is `null` or `undefined`?

File: packages/haul-babel-preset-react-native/src/index.ts (hunk 3 above)
Comment (zamotany) [reply packages/haul-babel-preset-react-native/src/index.ts]: It is but I cannot pass this value (`platform`) in a normal way as a preset option.

File: packages/haul-core/assets/debuggerWorker.js (hunk 4 above)
Comment (zamotany) [reply packages/haul-core/assets/debuggerWorker.js]: It does, it has to have some value or be `undefined`. `null` is incorrectly handled on iOS.

File: packages/haul-babel-preset-react-native/src/index.ts (hunk 3 above)
Comment (thymikee) [reply packages/haul-babel-preset-react-native/src/index.ts]: wtf Babel

Comment (zamotany) [reply packages/haul-babel-preset-react-native/src/index.ts]: I'm not sure it's babel fault, `platform` is like a dynamic value, we cannot hardcode it in the Babel config nor we can pass additional metadata to Babel plugins through `babel-loader`.

Comment (thymikee) [reply packages/haul-babel-preset-react-native/src/index.ts]: can you document it somehow somewhere?

**Pull request merged**